app/controllers/sessions_controller.rb

```diff
@@ -7,7 +7,12 @@ def create
     user = find_user
 
     if authenticated? user
-      handle_successful_login user
+      if user.activated?
+        handle_login user
+      else
+        flash[:warning] = t(".account_not_activated")
+        redirect_to root_path, status: :see_other
+      end
```

Comment:
```suggestion
      return handle_login(user) if user.activated?
      
      flash[:warning] = t(".account_not_activated")
      redirect_to root_path, status: :see_other
```
Còn lại logic bên trong này e viết cú pháp return if để clean code hơn nha